lib/Service/ConnectivityService.php

```diff
@@ -7,44 +7,92 @@
-		$output->writeln('<info>✓ Fetched /hosting/discovery endpoint</info>');
+		$output?->writeln('<info>✓ Fetched /hosting/discovery endpoint</info>');
 
 		$this->parser->getUrlSrcValue('application/vnd.openxmlformats-officedocument.wordprocessingml.document');
-		$output->writeln('<info>✓ Valid mimetype response</info>');
+		$output?->writeln('<info>✓ Valid mimetype response</info>');
 
 		// FIXME: Optional when allowing generic WOPI servers
 		$this->parser->getUrlSrcValue('Capabilities');
-		$output->writeln('<info>✓ Valid capabilities entry</info>');
+		$output?->writeln('<info>✓ Valid capabilities entry</info>');
 	}
 
-	public function testCapabilities(OutputInterface $output): void {
+	public function testCapabilities(?OutputInterface $output = null): void {
 		$this->capabilitiesService->resetCache();
 		$this->capabilitiesService->fetch();
-		$output->writeln('<info>✓ Fetched /hosting/capabilities endpoint</info>');
+		$output?->writeln('<info>✓ Fetched /hosting/capabilities endpoint</info>');
 
 		if ($this->capabilitiesService->getCapabilities() === []) {
 			throw new \Exception('Empty capabilities, unexpected result from ' . $this->capabilitiesService->getCapabilitiesEndpoint());
 		}
-		$output->writeln('<info>✓ Detected WOPI server: ' . $this->capabilitiesService->getServerProductName() . ' ' . $this->capabilitiesService->getProductVersion() . '</info>');
+		$output?->writeln('<info>✓ Detected WOPI server: ' . $this->capabilitiesService->getServerProductName() . ' ' . $this->capabilitiesService->getProductVersion() . '</info>');
+	}
+
+	public function testWopiAccess(?OutputInterface $output = null): void {
+		$client = $this->clientService->newClient();
+
+		if (!$this->capabilitiesService->hasWopiAccessCheck()) {
+			return;
+		}
+
+		$url = str_replace('/hosting/capabilities', '/hosting/wopiAccessCheck', $this->capabilitiesService->getCapabilitiesEndpoint());
+		$callbackUrl = $this->appConfig->getNextcloudUrl() ?: trim($this->urlGenerator->getAbsoluteURL(''), '/');
+
+		try {
+			$result = $client->post($url, ['body' => json_encode(['callbackUrl' => $callbackUrl . '/status.php']), 'headers' => ['Content-Type' => 'application/json']]);
+		} catch (ClientException $e) {
+			$result = $e->getResponse();
+		}
+		$response = json_decode($result->getBody(), true);
+
+		$errorMessage = match ($response['status']) {
+			'Ok' => null,
+			'NotHttpSuccess' => $this->l10n->t('The connection was successful but the response to the request was not 200'),
+			'HostNotFound' => $this->l10n->t('DNS error, the host is not known by the Collabora Online server'),
+			'WopiHostNotAllowed' => $this->l10n->t('The host for this request is not allowed to be used as a WOPI Host, this is likely a configuration issue in coolwsd.xml'),
+			'ConnectionAborted' => $this->l10n->t('The connection was aborted by the destination server'),
+			'CertificateValidation' => $this->l10n->t('The certificate of the response is invalid or otherwise not accepted'),
```

Comment:
There are two code that were added `SelfSignedCertificate` and `ExpiredCertificate`.

https://sdk.collaboraonline.com/docs/installation/Collabora_Online_Troubleshooting_Guide.html#checking-network-connectivity-between-collabora-online-and-the-wopi-host

Reply:
Thanks for checking and spotting, will add those

Reply:
Added